server panics with incorrect WOODPECKER_ENVIRONMENT

### Component

server

### Describe the bug

## Bug

The expected WOODPECKER_ENVIRONMENT is: 

```
WOODPECKER_ENVIRONMENT=k:v,foo:bar
```

But if we write: 
```
WOODPECKER_ENVIRONMENT=k:v,foo=bar
```

Then the server panics.

## Syntax

The syntax `WOODPECKER_ENVIRONMENT=k:v,foo:bar` is very difficult to read or write, especially when there are a lot of variables, or some variables contain comma or colon.

I think the new syntax (use 2 underlines) can be better:

```
WOODPECKER_ENV__X=y
WOODPECKER_ENV__FOO=bar
```

## Log

```
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/woodpecker-ci/woodpecker/server/plugins/environments.Filesystem(0xc0004cd780, 0x4, 0x4, 0xc0004cd780, 0x4)
        /woodpecker/src/github.com/woodpecker-ci/woodpecker/server/plugins/environments/filesystem.go:19 +0x2a8
main.setupEnvironService(0xc000240c80, 0x142d6c8, 0xc000194430, 0x7f2b240a3140, 0xc000194430)
        /woodpecker/src/github.com/woodpecker-ci/woodpecker/cmd/server/setup.go:176 +0x67
main.setupEvilGlobals(0xc000240c80, 0x142d6c8, 0xc000194430, 0x1425300, 0xc0006aa180)
        /woodpecker/src/github.com/woodpecker-ci/woodpecker/cmd/server/server.go:273 +0x45c
main.run(0xc000240c80, 0x0, 0x0)
        /woodpecker/src/github.com/woodpecker-ci/woodpecker/cmd/server/server.go:122 +0x413
github.com/urfave/cli/v2.(*App).RunContext(0xc0001a0680, 0x1418a68, 0xc000044040, 0xc000034220, 0x1, 0x1, 0x0, 0x0)
        /woodpecker/src/github.com/woodpecker-ci/woodpecker/vendor/github.com/urfave/cli/v2/app.go:322 +0x6fe
github.com/urfave/cli/v2.(*App).Run(...)
        /woodpecker/src/github.com/woodpecker-ci/woodpecker/vendor/github.com/urfave/cli/v2/app.go:224
main.main()
        /woodpecker/src/github.com/woodpecker-ci/woodpecker/cmd/server/main.go:35 +0x134
```

### System Info

```shell
Latest next.


"version": "next-8ce41788"
```


### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy).

--- a/cmd/server/setup.go
+++ b/cmd/server/setup.go
@@ -173,7 +173,7 @@ func setupRegistryService(c *cli.Context, s store.Store) model.RegistryService {
 }
 
 func setupEnvironService(c *cli.Context, s store.Store) model.EnvironService {
-	return environments.Filesystem(c.StringSlice("environment"))
+	return environments.Parse(c.StringSlice("environment"))
 }
 
 // setupRemote helper function to setup the remote from the CLI arguments.

--- a/server/plugins/environments/filesystem.go
+++ b/server/plugins/environments/filesystem.go
@@ -3,6 +3,8 @@ package environments
 import (
 	"strings"
 
+	"github.com/rs/zerolog/log"
+
 	"github.com/woodpecker-ci/woodpecker/server/model"
 )
 
@@ -10,14 +12,15 @@ type builtin struct {
 	globals []*model.Environ
 }
 
-// Filesystem returns a new local registry service.
-func Filesystem(params []string) model.EnvironService {
+// Parse returns a EnvironService based on a string slice where key and value are separated by a ":" delimeter.
+func Parse(params []string) model.EnvironService {
 	var globals []*model.Environ
 
 	for _, item := range params {
 		kvPair := strings.SplitN(item, ":", 2)
 		if len(kvPair) != 2 {
 			// ignore items only containing a key and no value
+			log.Warn().Msgf("key '%s' has no value, will be ignored", kvPair[0])
 			continue
 		}
 		globals = append(globals, &model.Environ{Name: kvPair[0], Value: kvPair[1]})

--- a/server/plugins/environments/filesystem.go
+++ b/server/plugins/environments/filesystem.go
@@ -16,6 +16,10 @@ func Filesystem(params []string) model.EnvironService {
 
 	for _, item := range params {
 		kvPair := strings.SplitN(item, ":", 2)
+		if len(kvPair) != 2 {
+			// ignore items only containing a key and no value
+			continue
+		}
 		globals = append(globals, &model.Environ{Name: kvPair[0], Value: kvPair[1]})
 	}
 	return &builtin{globals}

